website/static/js/oauthAddonNodeConfig.js

```diff
@@ -0,0 +1,249 @@
+        var $copyBtn = $(self.selector).find('.copyBtn');
+        new ZeroClipboard($copyBtn);
+    },
+    _updateCustomFields: function(settings){
+        this.validCredentials(settings.validCredentials);
+    },
+     /**
+     * Allows a user to create an access token from the nodeSettings page
+     */
+    connectAccount: function() {
+        this.options.connectAccount.call(this);
+    },
+    importAuth: function(){
+        var self = this;
+
+        self.updateAccounts(function() {
+            if (self.accounts().length > 1) {
+                bootbox.prompt({
+                    title: 'Choose ' + self.addonName + ' Access Token to Import',
+                    inputType: 'select',
+                    inputOptions: ko.utils.arrayMap(
+                        self.accounts(),
+                        function(item) {
+                            return {
+                                text: item.name,
+                                value: item.id
+                            };
+                        }
+                    ),
+                    value: self.accounts()[0].id,
+                    callback: (self.connectExistingAccount.bind(self))
+                });
+            } else {
+                bootbox.confirm({
+                    title: 'Import ' + self.addonName + ' Access Token?',
+                    message: self.messages.confirmAuth(),
+                    callback: function(confirmed) {
+                        if (confirmed) {
+                            self.connectExistingAccount.call(self, (self.accounts()[0].id));
+                        }
+                    }
+                });
+            }
+        });
+    }
+});
+
+OauthAddonFolderPickerViewModel.prototype.connectExistingAccount = function(account_id) {
+     var self = this;
+
+    $osf.putJSON(
+        self.urls().importAuth, {
+            external_account_id: account_id
+        }
+    ).done(self.onImportSuccess.bind(self)
+    ).fail(self.onImportError.bind(self));
+};
+
+OauthAddonFolderPickerViewModel.prototype.updateAccounts = function(callback) {
+    var self = this;
+    var request = $.get(self.urls().accounts);
+    request.done(function(data) {
+        self.accounts(data.accounts.map(function(account) {
+            return {
+                name: account.display_name,
+                id: account.id
+            };
+        }));
+        callback();
+    });
+    request.fail(function(xhr, textStatus, error) {
+        self.changeMessage(self.messages.UPDATE_ACCOUNTS_ERROR(), 'text-warning');
+        Raven.captureMessage('Could not GET ' + self.addonName + ' accounts for user', {
+            url: self.url,
+            textStatus: textStatus,
+            error: error
+        });
+    });
+};
+
+// Public API
+function OauthAddonNodeConfig(addonName, selector, url, folderPicker, opts) {
+    var self = this;
+    self.url = url;
+    self.folderPicker = folderPicker;
+    if (typeof opts === 'undefined') {
```

Comment:
Yes, probably, and probably, respectively. Additional options aren't necessarily required (depending on the addon), and forcing `{}` to be passed in worse than this. This file is based on `addonNodeConfig.js`, just designed for addons using the new oauth system, and this is how undefined options are handled there. Also, something like `opts = opts || {}` or a ternary will both raise a `ReferenceError` if `opts` is undefined